enclave/core/sgx/threadlocal.c

```diff
@@ -412,14 +336,24 @@ oe_result_t oe_thread_local_cleanup(oe_sgx_td_t* td)
         td->num_tls_atexit_functions = 0;
     }
 
-    /* Clear tls section if it exists */
+    /* Find the start of the tls section of all modules combined */
     uint8_t* fs = _get_fs_from_td(td);
-    uint8_t* tls_start = _get_thread_local_data_start(td);
-    // Invoke the cleanup function even when the thread-local data is empty
-    // (i.e., tls_start is NULL when tdata and tbss are zero).
+    int64_t tls_start_offset = 0;
+    for (size_t i = 0; i < OE_MAX_NUM_MODULES; ++i)
+    {
+        const oe_module_link_info_t* link_info = &oe_linked_modules[i];
+        if (oe_module_has_tls(link_info))
+            tls_start_offset =
+                oe_get_module_tls_start_offset(link_info, tls_start_offset);
+    }
+
+    // Invoke the cleanup function even when the thread-local data is
+    // empty (i.e., tls_start is NULL when tdata and tbss are zero).
     oe_allocator_thread_cleanup();
-    if (tls_start)
-        oe_memset_s(tls_start, (uint64_t)(fs - tls_start), 0, 0);
+
+    uint64_t tls_size = (size_t)-tls_start_offset;
```

Comment:
Probably want to abort if `tls_start_offset` is somehow positive. Also, if we're casting, it might as well be to `uint64_t` directly.

Reply:
Done. Also added oe_is_within_enclave checks just to be sure.